src/Gating/GateDataSet.cs

```diff
@@ -1781,6 +1811,38 @@ private static void ReadEndDate(GatingConfiguration.BaseGateType configurationGa
 		}
 
 
+		/// <summary>
+		/// Read the cloud contexts from a configuration gate
+		/// </summary>
+		/// <param name="configurationGate">configuration to read from</param>
+		/// <param name="gate">gate to add information to</param>
+		private void ReadCloudContexts(GatingConfiguration.BaseGateType configurationGate, Gate gate)
+		{
+			if (configurationGate.CloudContexts == null)
+			{
+				return;
+			}
+
+			gate.CloudContexts = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+			foreach (GatingConfiguration.BaseGateTypeCloudContext context in configurationGate.CloudContexts)
+			{
+				if (context.Name == "None")
+				{
+					gate.CloudContexts.Clear();
+					break;
+				}
+
+				string contextName = context.Name.ToString();
+				if (!gate.CloudContexts.Add(contextName))
+				{
+					ULSLogging.LogTraceTag(0, Categories.GateDataSet, Levels.Error,
```

Comment:
>Error [](start = 62, length = 5)

If this is an error, should we return an error? 

Or is it Ok to continue?

Reply:
I will change it to a Warning. It's when you write the same context in gate's XML twice